ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -4396,47 +4435,49 @@ let vm_migrate printer rpc session_id params =
             (read_map_params "vgpu" params)
         in
         let preferred_sr =
-          (* The preferred SR is determined to be as the SR that the destine host has a PDB attached to it,
-             and among the choices of that the shared is preferred first(as it is recommended to have shared storage
-             in pool to host VMs), and then the one with the maximum available space *)
+          (* The preferred SR is determined to be as the SR that the
+             destination host has a PDB attached to it, and among the choices
+             of that the shared is preferred first (as it is recommended to
+             have shared storage in pool to host VMs), and then the one with
+             the maximum available space *)
```

Comment:
Does this not suggest an algorithm that orders a list based on these criteria and picks the first element? Or would creating that list entail too many calls?

Reply:
That's right, the code now creates that list using 1 or 3 calls per SR.

Reply:
Now shared is only called once per element in the list, instead of twice.